pym/bob/cmds/build.py

```diff
@@ -1068,3 +1069,107 @@ def doClean(argv, bobRoot):
         if not args.dry_run:
             removePath(d)
 
+def doQueryPath(argv, bobRoot):
+    # Local imports
+    from string import Formatter
+
+    # Configure the parser
+    parser = argparse.ArgumentParser(prog="bob query-path",
+                                     description="Lists all packages and their last build directory. If a package has never been built, no output is shown for that package.")
+    parser.add_argument('packages', metavar='PACKAGE', type=str, nargs='+',
+        help="(Sub-)package to build")
+    parser.add_argument('-f', help='Select output format', default='{name}\t{dist}', metavar='FORMAT')
+    parser.add_argument('-D', default=[], action='append', dest="defines",
+        help="Override default environment variable")
+    parser.add_argument('-c', dest="configFile", default=[], action='append',
+        help="Use config File")
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--sandbox', action='store_true', help="Enable sandboxing")
+    group.add_argument('--no-sandbox', action='store_false', dest='sandbox', help="Disable sandboxing")
+    parser.set_defaults(sandbox=None)
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--develop', action='store_true',  dest='dev', help="Use developer mode")
+    group.add_argument('--release', action='store_false', dest='dev', help="Use release mode", default=True)
```

Comment:
I guess your intention was that developer mode is the default, right? It is also my preference. Unfortunately argparse applies the default only when a variable is defined for the first time. The way it is now the release mode is the default. So even though it looks silly the `default=True` must be one line above.